presto-hive/src/main/java/com/facebook/presto/hive/HiveSplitManager.java

```diff
@@ -178,11 +179,10 @@ public ConnectorSplitSource getSplits(ConnectorTransactionHandle transaction, Co
             return new FixedSplitSource(ImmutableList.of());
         }
 
-        // get buckets from first partition (arbitrary)
-        Optional<HiveBucketFilter> bucketFilter = layout.getBucketFilter();
+        Optional<HiveBucketFilter> bucketFilter = isBucketExecutionEnabled(session) ? layout.getBucketFilter() : Optional.empty();
```

Comment:
I think we should throw an exception  here: https://github.com/prestodb/presto/blob/cca4e4a628e3d7159ac029b8388acb55b44428dc/presto-hive/src/main/java/com/facebook/presto/hive/HivePartitionManager.java#L131 if bucketing execution is disabled (or table is non bucketed), and the bucket filter is present. After we have this check, i would remove this change, as `bucketFilter` will always be empty here if bucketHandle is empty.